.travis.yml

```diff
@@ -1,24 +1,40 @@
 sudo: false
 language: node_js
-node_js:
-- '8'
-- '10'
-- '11'
 cache:
   directories:
   - node_modules
 before_script:
 - yarn run lint
-script:
-- yarn run coverage
-- if [ -n "${FRONTEND-}" ]; then yarn run test-frontend ; fi
 after_script:
 - yarn run codecov
+stages:
+  - "Test"
+  - name: "Test frontend"
+    if: branch = master
 matrix:
   fast_finish: true
   include:
-    node_js: node
-    env: FRONTEND=true
+    - stage: "Test"
+      - node_js: 8
+      - script:
+        - yarn run coverage
```

Comment:
Lets move coverage as separate step. No need to run it on every version. So the stage can be combined like this. 

stages:
  - lint
  - test (run multiple node version(8,10,12) since it will run all unit tests)
  - test:frontend
  - coverage

With the above stages, you can do matrix now.

Reply:
I am still trying the combinations but the travis matrix is really poorly documented for the use case and `allow_failures` doesn't work as expected. Still figuring this out

Reply:
Let me give you an example

https://github.com/elastic/apm-agent-rum-js/blob/b57be6f7052233fc421f8699ad185d8e367ca57a/.travis.yml#L21-L61